librz/core/cmd_descs/cmd_print.yaml

```diff
@@ -435,6 +435,209 @@ commands:
+          - text: "c"
+            comment: "char (signed byte)"
+          - text: "C"
+            comment: "byte in decimal"
+          - text: "d"
+            comment: "0xHEX value (4 bytes) (see 'i' and 'x' formats)"
+          - text: "D"
+            comment: "disassemble one opcode"
+          - text: "e"
+            comment: "temporarily swap endian"
+          - text: "E"
+            comment: "resolve enum name (see t?)"
+          - text: "f"
+            comment: "float value (4 bytes)"
+          - text: "F"
+            comment: "double float value (8 bytes)"
+          - text: "i"
+            comment: "signed integer value (4 bytes) (see 'd' and 'x' formats)"
+          - text: "n"
+            comment: "next char specifies size of signed value (1, 2, 4, or 8 byte(s))"
+          - text: "N"
+            comment: "next char specifies size of unsigned value (1, 2, 4, or 8 byte(s))"
+          - text: "o"
+            comment: "octal value (4 bytes)"
+          - text: "p"
+            comment: "pointer reference (2, 4, or 8 bytes)"
+          - text: "q"
+            comment: "quadword (8 bytes)"
+          - text: "Q"
+            comment: "octoword (uint128_t) (16 bytes)"
+          - text: "r"
+            comment: "CPU register (`pf r (eax)plop`)"
+          - text: "s"
+            comment: "32 bit pointer to string (4 bytes)"
+          - text: "s"
+            comment: "32 bit pointer to string (4 bytes)"
+          - text: "t"
+            comment: "32 bit UNIX timestamp (4 bytes)"
+          - text: "T"
+            comment: "show ten first bytes of buffer"
+          - text: "u"
+            comment: "uleb128 (variable length)"
+          - text: "w"
+            comment: "word (2 bytes unsigned short in hex)"
+          - text: "x"
+            comment: "0xHEX value and flag (fd @ addr) (see 'd' and 'i' formats)"
+          - text: "X"
+            comment: "show formatted hexpairs"
+          - text: "z"
+            comment: "null terminated string"
+          - text: "Z"
+            comment: "null terminated wide string"
+          - text: "?"
+            comment: "data structure `pf ? (struct_name)example_name`"
+          - text: "*"
+            comment: "next char is pointer (honors 'asm.bits')"
+          - text: "+"
+            comment: "toggle show flags for each offset"
+          - text: ":"
+            comment: "skip 4 bytes"
+          - text: "."
+            comment: "skip 1 byte"
+          - text: ";"
+            comment: "rewind 4 bytes"
+          - text: ","
+            comment: "rewind 1 byte"
+      - name: Example of usages
+        entries:
+          - text: "pf '3xi foo bar'"
+            comment: "3-array of structures, each with named fields: 'foo' as hex, 'bar' as int"
+          - text: "pf 'B (BitFldType)arg_name'"
+            comment: "Resolve bitfield enum type for the arg_name"
+          - text: "pf 'E (EnumType)arg_name'"
```

Comment:
shouldn't we have each named argument as a separate argument of the `pf` command?

Reply:
Maybe in theory. But that would require rewriting almost all of the `librz/type/format.c`, which is an immense task in itself and is a story for another time. Moreover, it would make sense to rethink the format syntax/semantics before that task as well: https://github.com/rizinorg/rizin/issues/783